src/java/map/BinaryMask.java

```diff
@@ -93,6 +96,41 @@ public void set(int x, int y, boolean value) {
         mask[x][y] = value;
     }
 
+    public float getDistanceBetweenPoints(int x1, int y1, int x2, int y2) {
+        return (float) StrictMath.sqrt((double) ((x1 - x2) ^ 2 + (y1 - y2) ^ 2));
+    }
+
+    public float getDistanceBetweenPoints(int x1, int y1, Point Point2) {
+        return (float) StrictMath.sqrt((double) ((x1 - Point2.x) ^ 2 + (y1 - Point2.y) ^ 2));
+    }
+
+    public float getDistanceBetweenPoints(Point Point1, Point Point2) {
+        return (float) StrictMath.sqrt((double) ((Point1.x - Point2.x) ^ 2 + (Point2.x - Point2.y) ^ 2));
+    }
+
+    public Point getNearestPointInAreaTo(int x, int y) {
+        Point nearestPoint = new Point (x, y);
+        int rank;
+        FloatMask distanceField = getDistanceFieldFromPointInArea(x, y);
+        for (int a = 0; a < getSize(); a++) {
+            for (int b = 0; b < getSize(); b++) {
+                if (this.get(a, b)) {
+                    rank = distanceField.getRankOfPointInArea(a, b, this);
```

Comment:
why use rank instead of keeping track of the minimum distance? I am not sure rank guarantees you the minimum distance

Reply:
this will also cause you to loop through the mask again for every point. It would be more efficient to calculate the distance field once and then use it rather than calculate it every time

Reply:
if you use getDistanceField then you will get the distance of the closest true point so you will know what radius to check

Reply:
Keeping track of the minimum distance won't account for ties?
I also want to be able to get areas of set sizes with this.
I am open to suggestions that improve efficiency but don't reduce quality here.